docs/docs/core_plugins/index.html

```diff
@@ -43,36 +43,34 @@ <h1>Go Plugins</h1>
 <article>
     
 
-<h2 id="how-to-create-a-new-go-plugin">How to create a new Go plugin</h2>
+<h2 id="how-to-create-a-new-core-plugin">How to create a new core plugin</h2>
 
 <p>The <a href="https://github.com/puppetlabs/wash/tree/master/plugin">plugin</a> package defines a set of interfaces that a plugin can implement to enable specific behaviors.</p>
 
 <h3 id="starting-a-plugin">Starting a plugin</h3>
 
 <p>Create a new directory in <a href="https://github.com/puppetlabs/wash/tree/master/plugin">plugin</a> for the plugin.</p>
 
-<p>Create an object that implements the <code>Root</code> profile, as in it has:</p>
+<p>Create an object that implements the <code>Root</code> interface. This would typically be in a <code>root.go</code> file. See <a href="https://github.com/puppetlabs/wash/blob/master/plugin/docker/root.go">docker/root.go</a> for an example.</p>
 
-<ul>
-<li>a <code>Name</code> method to determine the name of the plugin when mounted</li>
-<li>an <code>Init</code> method that loads and validates credentials</li>
-<li>a <code>List</code> method that lists the first-level resources of the plugin</li>
-</ul>
+<p>NOTE: The <code>Init</code> method initializes the Root object&rsquo;s <code>EntryBase</code> configuration and any credentials.</p>
 
 <h3 id="extending-the-plugin">Extending the plugin</h3>
 
-<p>Add new types representing the types of resources the plugin can list.</p>
+<p>Each entry in the plugin&rsquo;s hierarchy should be a new type. This pattern&rsquo;s adopted by the existing core plugins (e.g. <a href="https://github.com/puppetlabs/wash/blob/master/plugin/aws/ec2Instance.go">ec2Instance</a> in AWS; <a href="https://github.com/puppetlabs/wash/blob/master/plugin/docker/container.go">container</a> in Docker). It is meant to make your plugin modular and easier to maintain.</p>
 
 <ul>
-<li>To make them appear as a directory, implement the <code>Group</code> interface.</li>
-<li>Resources with metadata should implement the <code>Resource</code> interface.</li>
-<li>To give files content, implement the <code>Readable</code> interface.</li>
-<li>Log-type files can implement the <code>Pipe</code> interface to expose a stream of new data.</li>
-<li>Files and directories that exist on a remote system or volume should implement the <code>File</code> interface. Additionally the <a href="https://github.com/puppetlabs/wash/tree/master/volume">volume</a> package provides helpers for representing these types.</li>
-<li>To support executing commands on a resource, implement the <code>Execable</code> interface.</li>
+<li>Entries with children (&ldquo;directories&rdquo;) should implement the <code>Parent</code> interface.</li>
```

Comment:
I prefer if we can mention `Entries` instead of `Resources` in the plugin tutorials just b/c that's more general and it opens the door for people who want to write more esoteric Wash plugins.